tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageSpecificChecks.cs

```diff
@@ -12,19 +10,242 @@ namespace Azure.Sdk.Tools.Cli.Services;
+                return new CLICheckResponse(1, "", $".NET SDK version {latestVersionNumber} is below minimum requirement of {RequiredDotNetVersion}");
+            }
+        }
+        else
+        {
+            _logger.LogError("Failed to parse .NET SDK version: {VersionString}", latestVersionNumber);
+            return new CLICheckResponse(1, "", $"Failed to parse .NET SDK version: {latestVersionNumber}");
+        }
+    }
+
+    private string? GetServiceDirectoryFromPath(string packagePath)
+    {
+        string? serviceDirectory = null;
+        var normalizedPath = packagePath.Replace('\\', '/');
+        var sdkIndex = normalizedPath.IndexOf("/sdk/", StringComparison.OrdinalIgnoreCase);
+
+        if (sdkIndex >= 0)
+        {
+            var pathAfterSdk = normalizedPath.Substring(sdkIndex + 5); // Skip "/sdk/"
+            var segments = pathAfterSdk.Split('/', StringSplitOptions.RemoveEmptyEntries);
+            if (segments.Length > 0)
+            {
+                serviceDirectory = segments[0];
+                _logger.LogDebug("Extracted service directory: {ServiceDirectory}", serviceDirectory);
+            }
+            else
+            {
+                _logger.LogDebug("No segments found after /sdk/ in path");
+            }
+        }
+        else
+        {
+            _logger.LogDebug("Path does not contain /sdk/ segment");
+        }
+
+        return serviceDirectory;
+    }
+
+    private string? GetPackageNameFromPath(string packagePath)
+    {
+        string? packageName = null;
+        var normalizedPath = packagePath.Replace('\\', '/');
+        var sdkIndex = normalizedPath.IndexOf("/sdk/", StringComparison.OrdinalIgnoreCase);
+
+        if (sdkIndex >= 0)
+        {
+            var pathAfterSdk = normalizedPath.Substring(sdkIndex + 5); // Skip "/sdk/"
+            var segments = pathAfterSdk.Split('/', StringSplitOptions.RemoveEmptyEntries);
+            if (segments.Length > 1)
+            {
+                packageName = segments[1];
+            }
+        }
+        return packageName;
+    }
+
+    private bool CheckAotCompatOptOut(string packagePath, string packageName)
+    {
+        try
+        {
+            var csprojFiles = Directory.GetFiles(packagePath, "*.csproj", SearchOption.AllDirectories);
+            var mainCsprojFile = csprojFiles.FirstOrDefault(f => 
+                Path.GetFileNameWithoutExtension(f).Equals(packageName, StringComparison.OrdinalIgnoreCase));
+            var csprojFile = mainCsprojFile ?? csprojFiles.FirstOrDefault();
+            
+            if (csprojFile == null)
+            {
+                _logger.LogDebug("No .csproj file found in package path: {PackagePath}", packagePath);
+                return false;
+            }
+
+            var projectContent = File.ReadAllText(csprojFile);
```

Comment:
Nit: We could make this method async and pass the cancellation token down to `ReadAllTextAsync`